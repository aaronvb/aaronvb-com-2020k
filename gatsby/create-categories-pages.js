'use strict';

const _ = require('lodash');
const path = require('path');

module.exports = async (graphql, actions) => {
  const { createPage } = actions;

  const result = await graphql(`
    {
      allMarkdownRemark(
        filter: { frontmatter: { template: { eq: "post" }, draft: { ne: true } } }
      ) {
        group(field: frontmatter___category) {
          fieldValue
          totalCount
        }
      }
    }
  `);

  _.each(result.data.allMarkdownRemark.group, (category) => {
    const categorySlug = `/category/${_.kebabCase(category.fieldValue)}`;

    createPage({
      path: categorySlug,
      component: path.resolve('./src/templates/category-template.js'),
      context: {
        category: category.fieldValue,
      }
    });
  });
};
