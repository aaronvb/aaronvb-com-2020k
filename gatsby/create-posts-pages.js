'use strict';

const path = require('path');

module.exports = async (graphql, actions) => {
  const { createPage } = actions;

  createPage({
    path: '/',
    component: path.resolve('./src/templates/index-template.js'),
  });
};
