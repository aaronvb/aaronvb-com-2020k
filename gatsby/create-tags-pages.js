'use strict';

const _ = require('lodash');
const path = require('path');

module.exports = async (graphql, actions) => {
  const { createPage } = actions;

  const result = await graphql(`
    {
      allMarkdownRemark(
        filter: { frontmatter: { template: { eq: "post" }, draft: { ne: true } } }
      ) {
        group(field: frontmatter___tags) {
          fieldValue
          totalCount
        }
      }
    }
  `);

  _.each(result.data.allMarkdownRemark.group, (tag) => {
    const tagSlug = `/tag/${_.kebabCase(tag.fieldValue)}`;

    createPage({
      path: tagSlug,
      component: path.resolve('./src/templates/tag-template.js'),
      context: {
        tag: tag.fieldValue,
      }
    });
  });
};
