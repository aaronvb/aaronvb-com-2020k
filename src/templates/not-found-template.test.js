// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import { useStaticQuery, StaticQuery } from 'gatsby';
import MockDate from 'mockdate';
import NotFoundTemplate from './not-found-template';
import siteMetadata from '../../jest/__fixtures__/site-metadata';
import type { RenderCallback } from '../types';

describe('NotFoundTemplate', () => {
  beforeEach(() => {
    StaticQuery.mockImplementationOnce(
      ({ render }: RenderCallback) => (
        render(siteMetadata)
      ),
      useStaticQuery.mockReturnValue(siteMetadata)
    );
  });

  it('renders correctly', () => {
    MockDate.set(new Date('2020-04-30T11:01:58.135Z'));
    const tree = renderer.create(<NotFoundTemplate />).toJSON();
    expect(tree).toMatchSnapshot();
    MockDate.reset();
  });
});
