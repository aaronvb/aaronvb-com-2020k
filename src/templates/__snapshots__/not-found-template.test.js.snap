// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundTemplate renders correctly 1`] = `
<div
  className="layout"
>
  <div
    className="topbar"
  >
    <div
      className="topbar__inner"
    >
      <div
        className="author"
      >
        <span
          className="author__title"
        >
          <a
            className="author__title-link"
            href="/"
          >
            Test name
          </a>
        </span>
      </div>
      <nav
        className="menu"
      >
        <a
          className="menu__list-item-link"
          href="//test/1/"
        >
          Test label 1
        </a>
        <a
          className="menu__list-item-link"
          href="//test/2/"
        >
          Test label 2
        </a>
        <a
          className="menu__list-item-link"
          href="//test/3/"
        >
          Test label 3
        </a>
      </nav>
    </div>
  </div>
  <div
    className="page"
  >
    <div
      className="page__inner"
    >
      <h4
        className="page__title"
      >
        NOT FOUND
      </h4>
      <div
        className="page__body"
      >
        <p>
          Sorry, the content you are trying to find is not here.
        </p>
        <p>
          If you feel like there should be something here, please 
          <a
            href="/#contact"
          >
            contact me
          </a>
          .
        </p>
      </div>
    </div>
  </div>
  <div
    className="footer"
  >
    <div
      className="footer__inner"
    >
      <div
        className="footer__copyright"
      >
        <div
          className="copyright"
        >
          Test copyright
        </div>
      </div>
      <div
        className="footer__daily"
      >
        <div
          className="daily"
        >
          Happy Thursday. â˜•
        </div>
      </div>
    </div>
  </div>
</div>
`;
